src/Mapbender/IntrospectionBundle/Component/WorkingSet.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+
+namespace Mapbender\IntrospectionBundle\Component;
+
+
+use Mapbender\CoreBundle\Entity\Application;
+use Mapbender\CoreBundle\Entity\Source;
+
+/**
+ * Set of objects to consider in various inspections.
+ *
+ * @todo: add elements
+ *
+ */
+class WorkingSet
+{
+    /** @var Application[] */
+    protected $applications;
+
+    /** @var  Source[] */
+    protected $sources;
+
+    public function __construct(array $applications = null, array $sources = null)
+    {
+        $this->setApplications($applications);
+        $this->setSources($sources);
+    }
+
+    /**
+     * @return Application[]
+     */
+    public function getApplications()
+    {
+        return $this->applications;
+    }
+
+    /**
+     * @return Source[]
+     */
+    public function getSources()
+    {
+        return $this->sources;
+    }
+
+    /**
+     * @param Application[]|null $applications
+     */
+    public function setApplications(array $applications = null)
```

Comment:
Do we actuallly need null here? May be just assume that Applications is always an array and skip the further checking?

Reply:
Actually, the entire WorkingSet thing doesn't seem as useful as hoped, because it's only checked to initialize a single entity list. Everything else is ignored.

It was easier to follow the defined Doctrine relations.

A better mechanism than this WorkingSet construct would be pluggable filters, that could e.g. strip disabled (or enabled) source instances, strip unpublished applications, or filter based on text matching the title.

That's why I kind of "forgot" to maintain the WorkingSet halfway through :p

Reply:
I think the original idea was to treat "null" as "do whatever you do by default" vs empty array = "you should really process the empty set of applications".
It's true that the implementation doesn't reflect that and it doesn't look worthwhile atm.